components/rating/stories/template.js

```diff
@@ -12,30 +12,89 @@ export const Template = ({
 	max = 5,
 	value = 0,
 	isReadOnly = false,
-	isFocused = false,
+	isKeyboardFocused = false,
 	isDisabled = true,
 	isEmphasized = false,
 	customClasses = [],
 	id = getRandomId("rating"),
 } = {}, context = {}) => {
 	const { updateArgs } = context;
+	document.addEventListener("DOMContentLoaded", function() {
+		const rating = document.getElementById(id);
+		if (!rating) return;
+		if (rating.classList.contains("is-disabled") || rating.classList.contains("is-readOnly")) return;
+		const icons = Array.from(rating.getElementsByClassName("spectrum-Rating-icon"));
+		let hoverIndex = -1;
+		let selectedIndex = -1;
+
+		const updateHoverState = () => {
+			icons.forEach((icon, index) => {
+				const activeStar = icon.querySelector(".spectrum-Rating-starActive");
+				const inactiveStar = icon.querySelector(".spectrum-Rating-starInactive");
+
+				if (index <= hoverIndex ||
+					(index <= selectedIndex && hoverIndex === -1)
+				) {
+					icon.classList.add("is-hoverSelection");
+					activeStar.style.display = "block";
+					inactiveStar.style.display = "none";
+				}
```

Comment:
Aren't `hoverIndex` and `selectedIndex` set to -1, which means that `index` will always be greater than them initially? It's working (don't get me wrong), but how is that working? I'm totally missing something here 🤦‍♀️